src/main/java/com/adobe/aem/compgenerator/utils/CommonUtils.java

```diff
@@ -251,4 +255,29 @@ private static BufferedWriter getFileWriterFromTemplate(String path,
         }
         return null;
     }
+    
+    
+    /**
+     * Gets the Tab properties if defined else returns all the properties.
+     *
+     * @param properties The {@link Property}
+     * @param tabs The {@link Tab}
+     * @return List<Property>
+     */
+    public static List<Property> getTabbedPropertiesIfExists(List<Property> properties, List<Tab> tabs) {
+        List<Property> updatedPropertiesList = null;
+        if (null != tabs && !tabs.isEmpty()) {
+            List<Property> sortedProperties = new ArrayList<>();
+            for (Tab tab : tabs) {
+                tab.getFields().stream()
```

Comment:
This code is going to put the fields into the order they are listed in the tab config rather than the order they are in up above, correct?   I'm somewhat concerned that it's inconsistent with how the functionality works in the case of no tabs, where fields are ordered in the way they are in the field configs.  However, I'm not convinced this is bad either as it's basically an "override" configuration...  I'm tentatively okay with this...

Reply:
There is always a possibility that we create the properties and then think about moving them b/w different tabs or change the order. As there are multiple use cases, i felt like keeping the ordering Tabs wise. If the tabs are not available, it should take the same order as properties. Please let me know otherwise